src/vs/workbench/contrib/debug/browser/debugSession.ts

```diff
@@ -103,11 +103,14 @@ export class DebugSession implements IDebugSession, IDisposable {
 			this.repl = (this.parentSession as DebugSession).repl;
 		}
 
-		const toDispose = this.rawListeners.add(new DisposableStore());
+		const toDispose = new DisposableStore();
 		const replListener = toDispose.add(new MutableDisposable());
 		replListener.value = this.repl.onDidChangeElements(() => this._onDidChangeREPLElements.fire());
 		if (lifecycleService) {
-			toDispose.add(lifecycleService.onWillShutdown(() => this.shutdown()));
+			toDispose.add(lifecycleService.onWillShutdown(() => {
```

Comment:
It looks like disposables made in this DebugSession hang around until the window is closed -- that's probably not right?

This change looks fine if it restores the status quo of 1.81 but we should probably have a better fix for this

Reply:
The DebugSession isn't disposed anyway- https://github.com/microsoft/vscode/issues/191904

Reply:
I've still got to fix that and make it work with the repl

Reply:
Sounds good. Looks like this (correctly) triggered the leak detector in some unit tests